### PR TITLE
[SYCL][DOC] Move experimental extension docs

### PR DESCRIPTION
**Pull request by gmlueck**
Move the experimental extensions to the
"sycl/doc/extensions/experimental" directory, and add a README
explaining the content of that directory.  Each extension document has
also been renamed to match the name of its feature-test macro.

Links to these extensions have been removed from the table in
"sycl/doc/extensions/README.md" because we want to avoid the
possibility of that table getting out-of-date with the directory
contents.  Instead, users can see the list of experimental extensions
by listing the content of the directory.

### CODE REVIEW

**Review by pvchupin — APPROVED** [review]

**Review by kbobrovs — DISMISSED** [review]

**Comment by gmlueck** [comment]
@bader: I resolved conflicts, which dismissed approvals from @kbobrovs and @pvchupin.  Can this be merged after checks complete, or do you need an approval also from @intel/dpcpp-specification-reviewers ?

**Comment by bader** [comment]
I can do either way. Technically to enable merge button **Required** CI checks must pass and code owners approve. I have additional privileges for this this repository and can merge even if some requirements are not satisfied.
Let me know what you'd like to me to do.

**Review by mkinsner — APPROVED** [review]

**Comment by lsatanov** [comment]
Hi. I've noticed that public doc links are not working after this change, e.g., like those found in [esimd doc](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/optimization-and-programming-guide/vectorization/explicit-vector-programming/explicit-simd-sycl-extension.html) page. Not sure who is responsible for those. Just FYI.

**Pull request merged**
